### PR TITLE
Block prefix ambiguity

### PR DESCRIPTION
**Pull request by jonny-no1**
The original pull request https://github.com/symfony/symfony-docs/pull/6636 accidentally targeted `symfony:master`. @stof

### CODE REVIEW

File: book/forms.rst [hunk 1]
```diff
@@ -1122,6 +1122,24 @@ the choice is ultimately up to you.
...
+    will be rendered incorrectly using the ``password_widget`` theme fragment.
+    You can still use the class name of your choice by overriding the
+    ``getBlockPrefix`` method of your form class::
+
+        public function getBlockPrefix()
+        {
+            return 'app_bundle_password';
+        }
+
+    Read the chapter ":doc:`/cookbook/form/create_custom_field_type.html`" for
```
Comment (xabbuh) [inline book/forms.rst]: You need to omit the actual extension (`.html`) here. Sphinx will take care of it when generating the documentation.

Comment (xabbuh) [inline book/forms.rst]: and I would move the word "chapter" after the link

**Comment by foaly-nr1** [comment]
@xabbuh Goodness, I tried to rebase but it seems I messed things up completely.

**Comment by xabbuh** [comment]
@foaly-nr1 Do you need help with rebasing your changes?

**Comment by foaly-nr1** [comment]
@xabbuh That would be most appreciated. If you could point out any resources I can check on how to fix this mess up?

**Comment by HeahDude** [comment]
`git rebase symfony/2.7 && git push -f` might help

**Comment by stof** [comment]
Actually, should be ``git rebase --onto symfony/2.7 symfony/master``, to rebase only commits which are not part of master. Otherwise you will rebase all commits of newer branches too

**Comment by foaly-nr1** [comment]
Phew, thanks @HeahDude and @stof. Sterling work!

File: book/forms.rst [hunk 2]
```diff
@@ -1128,6 +1128,24 @@ the choice is ultimately up to you.
             ));
         }
 
+.. caution::
+
+    When the name of your form class matches any registered field type that has
```
Comment (xabbuh) [inline book/forms.rst]: [...] form type class [...]

File: book/forms.rst [hunk 3]
```diff
@@ -1128,6 +1128,24 @@ the choice is ultimately up to you.
...
+    its own theme fragment, your form won't be rendered correctly. A form
```
Comment (xabbuh) [inline book/forms.rst]: A form type named [...]

**Comment by xabbuh** [comment]
👍

File: book/forms.rst [hunk 4]
```diff
@@ -1128,6 +1128,24 @@ the choice is ultimately up to you.
...
+    ``AppBundle\\Form\\Type\\PasswordType`` to change a user password for
+    instance will be mistaken for the built-in ``PasswordType`` field type and
+    will be rendered incorrectly using the ``password_widget`` theme fragment.
+    You can still use the class name of your choice by overriding the
+    ``getBlockPrefix`` method of your form class::
+
+        public function getBlockPrefix()
+        {
+            return 'app_bundle_password';
```
Comment (HeahDude) [inline book/forms.rst]: Why not just `app_password`?

**Comment by foaly-nr1** [comment]
@HeahDude @xabbuh done

**Comment by HeahDude** [comment]
👍
Status: Reviewed

**Comment by xabbuh** [comment]
ping @javiereguiluz @weaverryan @WouterJ

**Comment by WouterJ** [comment]
I think the caution makes much sense, I can imagine that it takes quite some time to figure this out yourself. However, I have some doubts about the targeted version and the location of this caution:

 * `getBlockPrefix()`, if I'm correct, is introduced in Symfony 2.8. Before this, the `FormTypeInterface#getName()` method had to return a unique name that was used for the Twig block. So it looks like this can only be an issue since Symfony 2.8?
 * This caution is placed in a section were we didn't talk about widget namings. It would make more sense to first talk about how templating blocks are named and then have this caution. Even more, I think it makes more sense to have this caution in the [How to Create a Custom Form Field Type](http://symfony.com/doc/current/cookbook/form/create_custom_field_type) cookbook article (more specifically, the "Creating a Template for the Field" section).

At last, I would propose to slightly change the wording. From *"A form type named `AppBundle\Form\Type\PasswordType` to change a user password for instance will be mistaken for the built-in PasswordType field type [...]"* to *"A form type named `AppBundle\Form\Type\PasswordType` to change a user's password for instance will have the same block name as the built-in `PasswordType` [...]".

Thank you very much for starting this PR @foaly-nr1! If you agree with my proposals, please update this PR if you have some time. Otherwise, feel free to comment.

**Comment by WouterJ** [comment]
Hi @foaly-nr1! I've taken your commit and applied my suggestions myself in https://github.com/symfony/symfony-docs/pull/6744 You still get the credits you deserve.

Thanks for starting this PR! I hope to see you back in the future :)

**Comment by weaverryan** [comment]
Other PR merged - thanks @foaly-nr1!

**Comment by foaly-nr1** [comment]
No problem @WouterJ. I'm on annual leave and abroad hence no reply

**Pull request closed without merging**